samples/android/video-recorder/src/org/opencv/samples/recorder/RecorderActivity.java

```diff
@@ -149,15 +158,16 @@ public Mat onCameraFrame(CvCameraViewFrame inputFrame)
         Log.d(TAG, "Camera frame arrived");
 
         Mat rgbMat = inputFrame.rgba();
-        Imgproc.cvtColor(rgbMat, rgbMat, Imgproc.COLOR_RGBA2RGB);
 
         int w = rgbMat.width();
         int h = rgbMat.height();
 
         Log.d(TAG, "Size: " + String.valueOf(w) + "x" + String.valueOf(h));
 
         if (videoWriter != null && videoWriter.isOpened()) {
-            videoWriter.write(rgbMat);
+            Mat frame = new Mat();
```

Comment:
It leads to memory leak. GC does not know about native part. Please make the variable class-wide and not re-create each time.